src/backend/http/course_service.rs

```diff
@@ -1,11 +1,27 @@
-use crate::{backend::storage::CourseStore, models::Course};
 use axum::{extract::Path, http::StatusCode, Extension, Json};
 
+use crate::backend::storage::{CourseStore, StorageError};
+use crate::models::Course;
+
+pub async fn get_course(
+    Path(course_code): Path<String>,
+    Extension(course_store): Extension<CourseStore>,
+) -> Result<Json<Course>, StatusCode> {
+    match course_store.select_course_by_code(&course_code).await {
+        Ok(course) => Ok(Json(course)),
+        Err(err) => match err {
+            StorageError::MissingRecords(_) => Err(StatusCode::NOT_FOUND),
+            _ => Err(StatusCode::INTERNAL_SERVER_ERROR),
+        },
+    }
+}
+
 pub async fn list_courses(
-    Path(course_name): Path<String>,
+    Path(course_code): Path<String>,
     Extension(course_store): Extension<CourseStore>,
 ) -> Result<Json<Vec<Course>>, StatusCode> {
-    match course_store.select_courses(&course_name).await {
+    // Current course offerings should only be listed from terms in the current year.
```

Comment:
Current year? or current term?

I think we can take some inspiration from the open api when designing this API:


<img width="1488" alt="image" src="https://github.com/GauravTalreja/ebb/assets/59036640/323e5ada-ef01-4c7b-b5cc-401bba0c991f">